components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/api/dto/RestError.java

```diff
@@ -65,5 +71,29 @@ public void setFields(List<String> fields) {
         this.fields = fields;
     }
 
+    @JsonSetter("statusCode")
+    void setStatusCode(final String statusCode) {
+        errorCode = statusCode;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof RestError)) {
+            return false;
+        }
+
+        final RestError other = (RestError) obj;
+
+        return Objects.equals(errorCode, other.errorCode) && Objects.equals(message, other.message) && Objects.equals(fields, other.fields);
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(new Object[] {errorCode, message, fields});
+    }
```

Comment:
Definitively not a big issue but when possible we should try to be gc friendly so I would enroll my own hash calculation to avoid creating a new array and an iterator (inside Arrays.hashCode) for each call